src/utils/internalLinks.js

```diff
@@ -120,7 +120,7 @@ export const decodeHashComponent = (string: string): string => {
 /** Parse the operand of a `stream` operator, returning a stream name. */
 const parseStreamOperand = (operand, streamsById): string => {
   // "New" (2018) format: ${stream_id}-${stream_name} .
-  const match = /^(\d+)-/.exec(operand);
+  const match = /^(\d+)(-.*)?/.exec(operand);
```

Comment:
nit: I would favor a "non-capturing group" instead of a "capturing group" (you can search [this doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Groups_and_Ranges) for those terms) for the added part, since we don't need to remember the part after the stream ID. (This should remain true as we move toward identifying streams with IDs, not names; that's #3918.)

Though, I notice that the web app [uses](https://github.com/showell/zulip/commit/c84f939d828274f69d69018fa99174e1ec0f0dd3#diff-e39c2df6624a5f0a813790e84aa1a9eed567fb0e70b1fa8df99b6e40d7c6ecedR289) a capturing group, hmm. @gnprice, what do you think; should we match that, just for consistency?